nautobot_floor_plan/forms.py

```diff
@@ -197,9 +197,34 @@ def __init__(self, *args, **kwargs):
 
         if self.instance.x_origin or self.instance.y_origin:
             if self.x_letters:
-                self.initial["x_origin"] = utils.grid_number_to_letter(self.instance.x_origin)
+                self.initial["x_origin"] = utils.axis_init_label_conversion(
+                    fp_obj.x_origin_seed,
+                    utils.grid_number_to_letter(self.instance.x_origin),
+                    fp_obj.x_axis_step,
+                    self.x_letters,
+                )
+            else:
+                self.initial["x_origin"] = utils.axis_init_label_conversion(
+                    fp_obj.x_origin_seed, self.initial["x_origin"], fp_obj.x_axis_step, self.x_letters
+                )
```

Comment:
I can't do this as a suggestion, but maybe replace the if/else with something optimized like this?
```python
              self.initial["x_origin"] = utils.axis_init_label_conversion(
                  fp_obj.x_origin_seed,
                  utils.grid_number_to_letter(self.instance.x_origin) if self.x_letters else self.initial["x_origin"],
                  fp_obj.x_axis_step,
                  self.x_letters,
              )
```